### PR TITLE
[Fix #7509] Layout/SpaceInsideArrayLiteralBrackets to correct empty lines

### LINKED ISSUE #7509: Layout/SpaceInsideArrayLiteralBrackets should correct empty lines
## Expected behavior

```rb
foo = [
]
foo
```

When I `rubocop -a`, rubocop corrects the empty line.
```rb
foo = []
foo
```

## Actual behavior
When I `rubocop -a`...

```rb
Offenses:

sample.rb:1:7: C: [Corrected] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside empty array brackets.
foo = [ ...
      ^

1 file inspected, 1 offense detected, 1 offense corrected
```
But it's not actually corrected and I keep getting this message.
I think this is a bug.

## Steps to reproduce the problem
```rb
$ vi sample.rb
foo = [
]
foo
```
```rb
$ rubocop -a sample.rb
Inspecting 1 file
C

Offenses:

sample.rb:1:7: C: [Corrected] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside empty array brackets.
foo = [ ...
      ^

1 file inspected, 1 offense detected, 1 offense corrected
```
But it's not corrected and I get the same message again.

```rb
$ rubocop -a sample.rb
Inspecting 1 file
C

Offenses:

sample.rb:1:7: C: [Corrected] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside empty array brackets.
foo = [ ...
      ^

1 file inspected, 1 offense detected, 1 offense corrected
```

## RuboCop version

```
$ rubocop -V
0.57.2 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-darwin16)
```

### PR DESCRIPTION
**Pull request by ayacai115**
Fixed Layout/SpaceInsideArrayLiteralBrackets to correct empty lines.
See https://github.com/rubocop-hq/rubocop/issues/7509 for details.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/correctors/space_corrector.rb [hunk 1]
```diff
@@ -12,12 +12,11 @@ class << self
         def empty_corrections(processed_source, corrector, empty_config,
                               left_token, right_token)
           @processed_source = processed_source
+          range = range_between(left_token.end_pos, right_token.begin_pos)
```
Comment (ayacai115) [inline lib/rubocop/cop/correctors/space_corrector.rb]: @buehmann Thanks for the suggestion!
https://github.com/rubocop-hq/rubocop/compare/master...buehmann:empty-array-literal/7509

**Comment by Drenmi** [comment]
Normally, the edge case that comes up for these kinds of cops is with comments, e.g.:

```ruby
foo = [
# Intentionally left empty
]
```

**Comment by ayacai115** [comment]
@Drenmi 
That's true.

What I'd like to fix is that rubocop raises a warning and says "corrected" but it's not actually corrected, which is a bug.

Regarding your suggestion, it's about how the syntax rule should be, so maybe we can discuss in another issue?

**Comment by Drenmi** [comment]
> What I'd like to fix is that rubocop raises a warning and says "corrected" but it's not actually corrected, which is a bug.

I see. 🙂 Yes. My comment is a bit tangential then, although fixing the bug is likely going to cause comments to be destroyed on auto-correct, so I think we should at least open a new issue for it, if its. 👍

**Comment by ayacai115** [comment]
@Drenmi Thanks for your suggestion😄 
I opened a issue for that: https://github.com/rubocop-hq/rubocop/issues/7512

**Comment by buehmann** [comment]
> Normally, the edge case that comes up for these kinds of cops is with comments, e.g.:

That's not an issue here because the "empty brackets" detection works on the token level.

**Comment by koic** [comment]
Thanks!

**Pull request merged**